cli/admin/src/main/java/io/pravega/cli/admin/bookkeeper/BookkeeperDeleteLedgersCommand.java

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright Pravega Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.cli.admin.bookkeeper;
+
+import io.pravega.cli.admin.CommandArgs;
+import io.pravega.segmentstore.storage.impl.bookkeeper.Ledgers;
+import lombok.Cleanup;
+import lombok.val;
+import org.apache.bookkeeper.client.BookKeeper;
+import org.apache.bookkeeper.client.api.ReadHandle;
+import org.apache.bookkeeper.conf.ClientConfiguration;
+import org.apache.bookkeeper.meta.LedgerManager;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * BookKeeper delete and recover command in case recovery fails with missing ledger(s).
+ */
+public class BookkeeperDeleteLedgersCommand  extends BookKeeperCommand {
+
+    public BookkeeperDeleteLedgersCommand(CommandArgs args) {
+        super(args);
+    }
+
+    @Override
+    public void execute() throws Exception {
+        ensureArgCount(2);
+        int logId = getIntArg(0);
+        long startId = getLongArg(1);
+
+        @Cleanup
+        val context = createContext();
+        @Cleanup
+        val log = context.logFactory.createDebugLogWrapper(logId);
+
+        // Display a summary of the BookKeeperLog.
+        val m = log.fetchMetadata();
+        if (m == null) {
+            output("BookKeeperLog '%s' does not exist.", logId);
+            return;
+        }
+
+        outputLogSummary(logId, m);
+
+        output("Ledgers will be permanently deleted from bookkeeper log '%s' " +
+                "starting with ledger id '%s' and reconcile will happen automatically ", logId, startId);
+        if (!confirmContinue()) {
+            output("Exiting delete-ledgers command");
+            return;
+        }
+        try {
+            log.deleteLedgersStartingWithId(startId);
+            output("Deleted ledgers from bookkeeper log '%s' starting with ledger id '%s' ", logId, startId);
+        } catch (Exception ex) {
+            output("Delete ledgers failed: Exiting " + ex.getMessage());
+            return;
+        }
+        output("Starting force over-write to recover log");
+        // over-write metadata
+        ClientConfiguration config = new ClientConfiguration()
```

Comment:
Question: why do we need to delete the ledgers from metadata, then delete the ledgers from Bookkeeper, and then reconcile again? Wouldn't it be shorter to do:
1. Disable the Bookkeeper Log (if not disabled already)
2. Delete all the ledgers from Bookkeeper Log beyond `ledger-id`.
3. Run reconciliation, which will leave the metadata with the existing ledgers.

That would be skipping the first step that this command is doing, right?

Reply:
Updated the pr as suggested.